src/controllers/DoctorLoadedPatientController.java

```diff
@@ -96,27 +96,39 @@ private Command DeletePatientPrescription() {
         };
     }
 
-/* PENDING IMPLEMENTATION IN PHASE 2
+
     private Command ViewPatientAppointments() {
         return (x) -> doctorView.viewAppointments(new AppointmentManager(getDatabase())
                 .getPatientAppointments(patientData));
     }
 
-    private Command getReport() {
-        return (x) -> new ReportManager(getDatabase()).getReportData(patientData);
+    private Command ViewPatientReports() {
+        return (x) -> {
+            ArrayList<ReportData> reportData = reportManager.getReportData(patientData);
+            if (reportData.size() > 0) {
+                doctorView.viewAllReports(reportData);
+                Integer targetReport = doctorView.viewReportPrompt();
+                doctorView.viewReport(reportData.get(targetReport - 1));
+            } else {
+                doctorView.noReports();
+            }
+        };
     }
 
-    private Command createReport() {
-        return (x) -> new ReportManager(getDatabase()).addReport(patientData, doctorData,
-                doctorView.reportDetailsPrompt().header(), doctorView.reportDetailsPrompt().body());
+    private Command CreatePatientReport() {
+        return (x) -> {
+            ReportDetails reportDetails = doctorView.reportDetailsPrompt();
+            reportManager.addReport(patientData, doctorData, reportDetails.header(), reportDetails.body());
+        };
     }
 
-    private Command deleteReport() {
+    private Command DeletePatientReport() {
         return (x) -> {
+            ArrayList<ReportData> reportData = reportManager.getReportData(patientData);
+
             Integer deleteIndex = doctorView.deleteReportPrompt(new ContactManager(getDatabase())
-                    .getContactData(patientData), new ReportManager(getDatabase()).getReportData(patientData));
-            getDatabase().getReportDatabase().remove(deleteIndex);
+                    .getContactData(patientData), reportData);
+            reportManager.deleteReport(reportData.get(deleteIndex));
```

Comment:
What happens when the index is out of range? The user should be made aware of his mistake. What happens when the user input is not an integer? The user should be made aware of his mistake. Look into the secretary deletion code for guidance. Furthermore, have you manually tested your code? Without even testing, this code clearly looks like it's prone to null pointer errors.

Reply:
I have manually tested my code and so far it works. I will however fix the presenter conventions and possible mistakes users could make.